pretix_eth/management/commands/confirm_payments.py

```diff
@@ -43,54 +47,97 @@ def handle(self, *args, **options):
-            )
-            full_id = order_payment.full_id
-
-            info = order_payment.info_data
-            token: IToken = all_token_and_network_ids_to_tokens[info["currency_type"]]
-            expected_network_id = token.NETWORK_IDENTIFIER
-            expected_network_rpc_url_key = f"{expected_network_id}_RPC_URL"
-            network_rpc_url = None
-
-            if expected_network_rpc_url_key in rpc_urls:
-                network_rpc_url = rpc_urls[expected_network_rpc_url_key]
-            else:
-                logger.warning(
-                    f"No RPC URL configured for {expected_network_id}. Skipping..."
+        with scope(organizer=event.organizer):
+            unconfirmed_order_payments = OrderPayment.objects.filter(
+                order__event=event,
+                state__in=(
+                    OrderPayment.PAYMENT_STATE_CREATED,
+                    OrderPayment.PAYMENT_STATE_PENDING
                 )
-                continue
+            )
 
-            expected_amount = info["amount"]
+        for order_payment in unconfirmed_order_payments:
+            # it is tempting to put .filter(invalid=False) here, but remember
+            # there is still a chance that low-gas txs are mined later on.
+            for signed_message in order_payment.signed_messages.all():
+                rpc_urls = json.loads(
+                    order_payment.payment_provider.settings.NETWORK_RPC_URL
+                )
+                full_id = order_payment.full_id
 
-            # Get balance.
-            balance = token.get_balance_of_address(hex_address, network_rpc_url)
+                info = order_payment.info_data
+                token: IToken = all_token_and_network_ids_to_tokens[info["currency_type"]]
+                expected_network_id = token.NETWORK_IDENTIFIER
+                expected_network_rpc_url_key = f"{expected_network_id}_RPC_URL"
 
-            if balance > 0:
-                logger.info(f"Payments found for {full_id} at {hex_address}:")
-                if balance < expected_amount:
+                if expected_network_rpc_url_key in rpc_urls:
+                    network_rpc_url = rpc_urls[expected_network_rpc_url_key]
+                else:
                     logger.warning(
-                        f"  * Expected payment of at least {expected_amount} {token.TOKEN_SYMBOL}"
+                        f"No RPC URL configured for {expected_network_id}. Skipping..."
                     )
+                    continue
+
+                expected_amount = info["amount"]
+
+                # Get balance
+                w3 = Web3(load_provider_from_uri(network_rpc_url))
+                # native asset
+                if token.IS_NATIVE_ASSET:
+                    try:
+                        transaction_details = w3.eth.getTransaction(signed_message.transaction_hash)
+                    except TransactionNotFound:
+                        if signed_message.age > 30*60:
```

Comment:
can this be a config? The amount of time before we invalidate a transaction?

Reply:
It's a config now! This is actually the amount of time before a customer can try to pay again in case their first payment didn't confirm.